Modules/posixmodule.c

```diff
@@ -5553,6 +5520,103 @@ os__path_normpath_impl(PyObject *module, path_t *path)
     return result;
 }
 
+/*[clinic input]
+os._path_abspath
+
+    path: path_t(make_wide=True, nonstrict=True)
+    /
+
+Return an absolute path.
+[clinic start generated code]*/
+
+static PyObject *
+os__path_abspath_impl(PyObject *module, path_t *path)
+/*[clinic end generated code: output=bb40fbf3be7251a4 input=8870bbcaa7c920a1]*/
+{
+    Py_ssize_t abs_len;
+    wchar_t *abs, *abs_buf = NULL, *cwd_buf = NULL;
```

Comment:
`cwd_buf` is only used for the POSIX path; put it in that scope, for improved code clarity.